client/trino-cli/pom.xml

```diff
@@ -81,6 +81,12 @@
             <artifactId>modernizer-maven-annotations</artifactId>
         </dependency>
 
+        <dependency>
+            <groupId>org.jline</groupId>
+            <artifactId>jline-builtins</artifactId>
```

Comment:
Why not use `cat`-equivalent as the fallback instead of adding dependency?

Reply:
`cat` is not a pager, query results would be printed in the same way without `cat`.

Reply:
> `cat` is not a pager

yes, i know.

i think the goal here is to make CLI work in an environment where `less` is not present



> My _actual_ use case is that I'm trying to build a fully static version of the CLI using GraalVM and put it inside a scratch-based Docker container that doesn't have any other files, just a single binary.

that's a different goal than the one i assumed above.

you not only want the CLI not to fail miserably when `less` is not present, you actually want it to behave _reasonably_.

Would it be possible for your custom build to explicitly opt-in into a different behavior (built-in pager), without relying on the fallback logic. I think the fallback logic should be simpler than it is, but this wouldn't perhaps match your needs for the isolated image.

Reply:
> The intention of this PR is to provide a built-in alternative, without changing any default behavior. I was thinking about an opt-in flag, but it seems to unnecessarily complicate things, giving users too many choices is bad UX.

I meant an opt-in on the packaging level, not a command line switch.

Your custom image knows it doesn't ship `less`, so it also could know not to even try finding it on PATH.

Reply:
Interesting idea, but I've never done that and need to do some research first.